File: typescript/Two Pointers/shift_zeros_to_the_end_naive.ts
function shiftZerosToTheEndNaive(nums: number[]): void {
    let temp: number[] = Array(nums.length).fill(0);
    let i = 0;
    // Add all non-zero elements to the left of 'temp'.
    for (let number of nums) {
        if (number !== 0) {
            temp[i] = number;
            i++;
        }
    }
    // Set 'nums' to 'temp'.
    for (let j = 0; j < nums.length; j++) {
        nums[j] = temp[j];
    }
}

File: rust/Two Pointers/shift_zeros_to_the_end_naive.rs
fn shift_zeros_to_the_end_naive(nums: &mut Vec<i32>) {
    let mut temp = vec![0; nums.len()];
    let mut i = 0;

    // Add all non-zero elements to the left of 'temp'.
    for num in nums.iter() {
        if *num != 0 {
            temp[i] = *num;
            i += 1;
        }
    }

    // Set 'nums' to 'temp'.
    for j in 0..nums.len() {
        nums[j] = temp[j];
    }
}
